airgun/entities/contenthost.py

```diff
@@ -270,3 +278,27 @@ def step(self, *args, **kwargs):
         environment = kwargs.get('environment')
         self.parent.errata.search(errata_id, lce=environment)
         self.parent.errata.table.row(id=errata_id)['Id'].widget.click()
+
+
+@navigator.register(ContentHostEntity, 'Details')
```

Comment:
I think this modification needs to be added in the https://github.com/SatelliteQE/airgun/pull/850/files#diff-db1d0b0ef7fa44fdeed46d3c6c6432bb84e46257346e0a5a9c37cde62641bff4R243 line/code block. One another approach could be applied to the prerequisite such that the navigation happens from the new host details page.   

e.g.

``` python
@navigator.register(ContentHostEntity, 'NewEdit')
class EditNewContentHost(NavigateStep):
    """ Navigate to the new Content Host details screen.
    Args:
        entity_name: name of the content host
    """
    VIEW = NewHostDetailsView
    def prerequisite(self, *args, **kwargs):
        return self.navigate_to(self.obj, 'All')
    def step(self, *args, **kwargs):
        entity_name = kwargs.get('entity_name')
        self.parent.search(entity_name)
        self.parent.table.row(name=entity_name)['Name'].widget.click()

  
@navigator.register(ContentHostEntity, 'OldEdit')
class EditOldContentHost(NavigateStep):
    """ Navigate to the Old Content Host details screen.
    Args:
        entity_name: name of the content host
    """
    VIEW = ContentHostDetailsView
    def prerequisite(self, *args, **kwargs):
        entity_name = kwargs.get('entity_name')
        return self.navigate_to(self.obj, 'NewEdit', entity_name)
    def step(self, *args, **kwargs):
        self.parent.dropdown.item_select('Legacy content host UI')
```

Reply:
> I'm even okay with the implementation you added, I still think this needs to be inside `Edit`. We should have a way to access that with the read method rather than adding an extra method as `get_details`

@omkarkhatavkar Hello Omkar, I have researched a long time ,but  I still found the way to read the the newhostcontentview to oldhostcontentview. Also I found there are some other cases the same realization as this PR.
Can you give me some help about this? If there is no resolution for the way you sugggested, could you please accept the update the merge the PR? Because this PR influence many of the virt-who UI cases. Thank you very much.

Reply:
I ack, please addressed another comment I gave. we will merge this quiickly once this change is updated

Reply:
> I ack, please addressed another comment I gave. we will merge this quiickly once this change is updated

robottelo PR https://github.com/SatelliteQE/robottelo/pull/11543/commits/d904e7767d3e9c7258fd80bee7eb5308993121e2
 have been also updated